### ISSUE TITLE
Voice building

### ISSUE THREAD

**Issue by elnursamedzade** [marytts/gradle-marytts-voicebuilding-plugin#127]
Hello.
Please, can you send me a build.gradle sample? I want to understand what I'm doing wrong.
I am a visually impaired user. I am using Nvda screen reader. I want to use my own voice as a synthesizer. Please feel free to contact me privately. I need your help a lot. My e-mail address is elnur.samedzade@gmail.com
Thank you.

**Comment by psibre** [comment]
Thanks for the context! We're in the process of updating our voices to work with the latest code, but it takes some time. I'll try to prioritize marytts/voice-dfki-ot#3, which will hopefully help you get more experience with how it's all supposed to work. =)

**Comment by elnursamedzade** [comment]
Thank you very much!
Isn't there any way I can add my own data to the voice-dfki-ot voice right now? What codes do I need to change in the build.gradle file? I am replacing wav files with my own voice but md5 gives encryption error. I don't want to leave Marytts. I've been fighting for days. I prepared my own text files, recorded my voice. I don't want to leave now.

**Comment by psibre** [comment]
There's some fairly complicated build logic in the current build script for voice-dfki-ot:
https://github.com/marytts/voice-dfki-ot/blob/master/build.gradle#L32-L116

This was intended to be streamlined by using a [common plugin](https://github.com/m2ci-msp/gradle-flaml-plugin), like in marytts/voice-enst-camille#5, but back at the end of 2018, there wasn't enough time, and then I switched jobs. I only recently started to free up enough time to help maintain MaryTTS a bit more here and there.

## TL;DR

1. You can wait another day or two, then the voice-dfki-ot project should be updated and easier to understand, *or*
1. You can hack the voice-dfki-ot build logic on your own:
    1. Copy your voice data (wav, text, and lab files) into custom directories called `my_wav_files`, `my_text_files`, and `my_lab_files`, respectively
    1. Use Gradle v4.10.3
    1. Patch the build logic by adapting something like this:
```diff
diff --git a/build.gradle b/build.gradle
index a6d21fc..13ebc60 100644
--- a/build.gradle
+++ b/build.gradle
@@ -8,7 +8,7 @@ buildscript {
 }
 
 plugins {
-    id 'de.dfki.mary.voicebuilding-legacy' version '5.2.2'
+    id 'de.dfki.mary.voicebuilding-legacy' version '5.3.2'
     id 'de.undercouch.download' version '3.1.1'
 }
 
@@ -115,8 +115,17 @@ task extractWav {
     }
 }
 
+wav.srcFiles = fileTree('my_wav_files').include('*.wav')
+
+generateAllophones.srcDir = file('my_text_files')
+
+task copyLabFiles(type: Copy) {
+    from 'my_lab_files'
+    into "$buildDir/lab"
+}
+
 legacyInit {
-    dependsOn extractLab, extractText, extractWav
+    dependsOn wav, copyLabFiles
     def basenameListFile = file("$buildDir/basenames.lst")
     outputs.files basenameListFile
     doLast {
@@ -124,37 +133,8 @@ legacyInit {
             fileTree(wav.destDir).collect {
                 it.name - '.wav'
             }.minus([
-                    'ot0044',
-                    'ot0100',
-                    'ot0103',
-                    'ot0256',
-                    'ot0282',
-                    'ot0291',
-                    'ot0307',
-                    'ot0323',
-                    'ot0370',
-                    'ot0453',
-                    'ot0497',
-                    'ot0538',
-                    'ot0569',
-                    'ot0600',
-                    'ot0662',
-                    'ot0673',
-                    'ot0745',
-                    'ot0746',
-                    'ot0774',
-                    'ot0791',
-                    'ot0818',
-                    'ot0853',
-                    'ot0882',
-                    'ot0883',
-                    'ot0893',
-                    'ot0918',
-                    'ot0944',
-                    'ot1073',
-                    'ot1106',
-                    'ot1132',
-                    'ot1159'
+                    // list here basenames with invalid data
+                    // in case of processing errors
             ]).each {
                 basenameList.writeLine it
             }
```
But if you run into problems here, please be patient until we update the build logic properly.

**Comment by elnursamedzade** [comment]
Hello mr Psibre.
Unfortunately, it did not happen. I will be patient and wait for your return. I hope one day I will be a marytts user and leave happy here. I wish you good work. Please keep me updated. Best regards.

**Comment by psibre** [comment]
@elnursamedzade Can we close this issue and continue discussion in marytts/voice-dfki-ot#3?
